test/client/fixtures/utils/dom-test.js

```diff
@@ -131,10 +131,39 @@ test('isWindow', function () {
     };
 
     ok(domUtils.isWindow(window));
+    ok(!domUtils.isWindow([
+        {
+            toString: function () {
+                ok(false);
+            }
+        }
+    ]));
 
     window.toString = storedToString;
 });
 
+asyncTest('isWindow for a cross-domain window', function () {
+    var iframe = document.createElement('iframe');
+
+    iframe.id  = 'test' + Date.now();
+    iframe.src = window.getCrossDomainPageUrl('../../data/cross-domain/simple-page.html');
+
+    window.QUnitGlobals.waitForIframe(iframe)
+        .then(function () {
+            var iframeWindow = iframe.contentWindow;
+
+            ok(domUtils.isWindow(iframeWindow));
+
+            // NOTE: The firefox does not provide access to the cross-domain location.
+            if (!browserUtils.isFirefox)
+                ok(!domUtils.isWindow(iframeWindow.location));
```

Comment:
I think we should move `!browserUtils.isFirefox condition before test and remove `ok(!domUtils.isWindow(iframeWindow.location));`.
Because you test checks `window`.